testutil/compose/config.go

```diff
@@ -45,43 +42,53 @@ const (
 	vcLighthouse vcType = "lighthouse"
 )
 
-// keyGen defines a key generation process.
-type keyGen string
+// KeyGen defines a key generation process.
+type KeyGen string
 
 const (
-	keyGenDKG    keyGen = "dkg"
-	keyGenCreate keyGen = "create"
-	keyGenSplit  keyGen = "split"
+	keyGenDKG    KeyGen = "dkg"
+	keyGenCreate KeyGen = "create"
+	keyGenSplit  KeyGen = "split"
 )
 
 // config defines a local compose cluster; including both keygen and running a cluster.
 type config struct {
 	// Version defines the compose config version.
 	Version string `json:"version"`
 
+	// NumNodes is the number of charon nodes in the cluster.
+	NumNodes int
+
+	// Threshold required for signature reconstruction. Defaults to safe value for number of nodes/peers.
+	Threshold int
+
+	// NumValidators is the number of DVs (n*32ETH) to be created in the cluster lock file.
+	NumValidators int
+
 	// ImageTag defines the charon docker image tag: ghcr.io/obolnetwork/charon:{ImageTag}.
 	ImageTag string `json:"image_tag"`
 
 	// VCs define the types of validator clients to use.
 	VCs []vcType `json:"validator_clients"`
 
-	// keyGen defines the key generation process.
-	KeyGen keyGen `json:"key_gen"`
+	// KeyGen defines the key generation process.
+	KeyGen KeyGen `json:"key_gen"`
 
 	// BeaconNode url endpoint or "mock" for simnet.
 	BeaconNode string `json:"beacon_node"`
-
-	// Def is the cluster definition.
-	Def cluster.Definition `json:"definition"`
```

Comment:
cluster definition (and all other charon dependencies) completely removed. Only using docker-compose as API.